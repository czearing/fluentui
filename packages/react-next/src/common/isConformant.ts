import { isConformant as baseIsConformant, IsConformantOptions } from '@fluentui/react-conformance';

export function isConformant(
  testInfo: Omit<IsConformantOptions, 'componentPath' | 'packageVersion'> & { componentPath?: string },
) {
  const defaultOptions = {
    disabledTests: ['has-docblock', 'kebab-aria-attributes'],
    componentPath: module!.parent!.filename.replace('.test', ''),
    packageVersion: '@fluentui/react-next',
  };

  baseIsConformant(defaultOptions, testInfo);
}
