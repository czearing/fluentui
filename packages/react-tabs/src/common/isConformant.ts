import { isConformant as baseIsConformant, IsConformantOptions } from '@fluentui/react-conformance';

export function isConformant(testInfo: Omit<IsConformantOptions, 'componentPath' | 'packageVersion'>) {
  const defaultOptions = {
    disabledTests: ['has-docblock', 'kebab-aria-attributes'],
    componentPath: module!.parent!.filename.replace('.test', ''),
    packageVersion: '@fluentui/react-tabs',
  };

  baseIsConformant(defaultOptions, testInfo);
}
