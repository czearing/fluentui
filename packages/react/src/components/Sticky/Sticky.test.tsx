import * as React from 'react';
import { Sticky } from './Sticky';
import { safeCreate, safeMount } from '@fluentui/test-utilities';
import { resetIds } from '@fluentui/utilities';
import { isConformant } from '../../common/isConformant';

const ReactDOM = require('react-dom');

describe('Sticky', () => {
  const createPortal = ReactDOM.createPortal;

  beforeEach(() => {
    resetIds();
  });

  afterEach(() => {
    ReactDOM.createPortal = createPortal;
  });

  // Conformance Tests:
  isConformant({
    Component: Sticky,
    displayName: 'Sticky',
    // Problem: Ref doesn't match DOM node and returns outermost div.
    // Solution: Ensure ref is passed correctly to the root element.
    disabledTests: ['component-handles-ref', 'component-has-root-ref', 'component-handles-classname'],
  });

  // Snapshot Tests:
  it('renders Sticky (correctly)', () => {
    ReactDOM.createPortal = jest.fn(element => element);

    safeCreate(<Sticky>This is a test</Sticky>, component => {
      const tree = component!.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
