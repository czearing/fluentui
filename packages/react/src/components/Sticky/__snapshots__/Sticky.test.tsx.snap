// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticky renders Sticky (correctly) 1`] = `
<div>
  This is a test
</div>
`;
